pkg/apis/core/validation/validation.go

```diff
@@ -5580,6 +5580,11 @@ func ValidateNamespace(namespace *core.Namespace) field.ErrorList {
 	for i := range namespace.Spec.Finalizers {
 		allErrs = append(allErrs, validateFinalizerName(string(namespace.Spec.Finalizers[i]), field.NewPath("spec", "finalizers"))...)
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
```

Comment:
shouldn't here be something like if utilfeature.Default.Enabled (instead of if NOT utilfeature.Default.Enabled)?

And maybe change also the error message to something like `fmt.Sprintf("label cannot be changed")))`